website_sale_product_contract_gift/models/account_move.py

```diff
@@ -0,0 +1,31 @@
+# SPDX-FileCopyrightText: 2024 Coop IT Easy SC
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+from odoo import models
+from odoo.tools import email_normalize
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def create_user_for_gift(self):
+        for invoice in self:
+            partner_id = invoice.partner_id
+            is_gift = any(
+                invoice.line_ids.mapped("contract_line_id")
+                .mapped("contract_id")
+                .mapped("is_gift")
```

Comment:
in 16.0, this should work:
```suggestion
                invoice.line_ids.contract_line_id.contract_id
                .mapped("is_gift")
```

Reply:
done.